drivers/spi/spi_context.h

```diff
@@ -311,13 +321,21 @@ void spi_context_update_tx(struct spi_context *ctx, uint8_t dfs, uint32_t len)
 
 	ctx->tx_len -= len;
 	if (!ctx->tx_len) {
-		ctx->tx_count--;
-		if (ctx->tx_count) {
+		/* This loop skips zero-length buffers in the set, if any. */
+		while (1) {
+			ctx->tx_count--;
+			if (!ctx->tx_count) {
+				ctx->tx_buf = NULL;
+				break;
+			}
+
 			ctx->current_tx++;
-			ctx->tx_buf = (const uint8_t *)ctx->current_tx->buf;
-			ctx->tx_len = ctx->current_tx->len / dfs;
-		} else {
-			ctx->tx_buf = NULL;
+			if ((ctx->current_tx->len / dfs) != 0) {
+				ctx->tx_len = ctx->current_tx->len / dfs;
+				ctx->tx_buf =
+					(const uint8_t *)ctx->current_tx->buf;
+				break;
+			}
```

Comment:
This while-loop here implements the same thing as the white loop in `spi_context_buffers_setup()`, but in a slightly different way. Could create a helper function to skip zero-length buffers, that would be used from `spi_context_update_tx()` and `spi_context_buffers_setup()`?

Another thing that I noticed is that RX and TX handling look very similar. Ideally those would also be deduplicated, as there is a big chance of introducing copy-paste bug when handling rx and tx. Maybe this is not a thing for this PR specifically (though it would be really nice to have), but what about a follow-up PR?

Overall this looks fine to me, the only thing that bugs me is that there are 4 places of really similar code, which takes 4x times longer to read, review and debug.

Reply:
I refactored the implementation so that the common part of those 4 pieces of code is extracted into a helper function. Could you take one more look?

Reply:
Much better now, thanks!